discovery/kubernetes_api_discovery.go

```diff
@@ -52,6 +52,12 @@ func (k *K8sAPIDiscoverer) Services() []service.Service {
 		hostname, ip := getIPHostForNode(&node)
 
 		for _, item := range k.discoveredSvcs.Items {
+			// We require an annotation called 'ServiceName' to make sure this is
+			// a service we want to announce.
+			if item.Metadata.Labels.ServiceName == "" {
```

Comment:
note: we have 26 services that do not have a service name in their deploy-config file